### PR TITLE
go.mod:update tidb & tikv-client to new version

### PR DESCRIPTION
**Pull request by AndreMouche**
Signed-off-by: shirly <AndreMouche@126.com>

### What problem does this PR solve? 

This PR updates [tidb](https://github.com/pingcap/tidb) & [go-client](https://github.com/tikv/client-go) to the new version. 
### Check List 

 - Unit test

### Release Note

- No release note

### CODE REVIEW

**Review by disksing — APPROVED** [review]

**Review by overvenus — APPROVED** [review]

**Comment by overvenus** [comment]
Find a panic in tests

https://ci.pingcap.net/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/6172/pipeline#step-471-log-68

```
[2021-07-27T03:30:45.315Z] panic: runtime error: invalid memory address or nil pointer dereference

[2021-07-27T03:30:45.316Z] [signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x2eaded2]

[2021-07-27T03:30:45.316Z] 

[2021-07-27T03:30:45.316Z] goroutine 778 [running]:

[2021-07-27T03:30:45.316Z] github.com/pingcap/br/pkg/lightning/backend/kv.(*session).GetBuiltinFunctionUsage(0xc00057c080, 0xc00057c080)

[2021-07-27T03:30:45.316Z] 	<autogenerated>:1 +0x32

[2021-07-27T03:30:45.316Z] github.com/pingcap/tidb/planner/core.(*expressionRewriter).newFunction(0xc001584340, 0xc000fc4733, 0x4, 0xc00024a0f8, 0xc00064c040, 0x3, 0x4, 0xc000140fc8, 0x7effe4007558, 0x20, ...)

[2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:1194 +0x173

[2021-07-27T03:30:45.316Z] github.com/pingcap/tidb/planner/core.(*expressionRewriter).funcCallToExpression(0xc001584340, 0xc00024a0e0)

[2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:1752 +0x8db

[2021-07-27T03:30:45.316Z] github.com/pingcap/tidb/planner/core.(*expressionRewriter).Leave(0xc001584340, 0x406fdf0, 0xc00024a0e0, 0x40884a8, 0xc000c169c0, 0x1)

[2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:1064 +0x19ad

[2021-07-27T03:30:45.316Z] github.com/pingcap/parser/ast.(*FuncCallExpr).Accept(0xc00024a0e0, 0x40361b8, 0xc001584340, 0x406fdf0, 0xc00024a2a0, 0xc00168a000)

[2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/parser@v0.0.0-20210707071004-31c87e37af5c/ast/functions.go:530 +0x18e

[2021-07-27T03:30:45.316Z] github.com/pingcap/parser/ast.(*FuncCallExpr).Accept(0xc00024a2a0, 0x40361b8, 0xc001584340, 0x409b010, 0xc001584000, 0xc0001413e8)

[2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/parser@v0.0.0-20210707071004-31c87e37af5c/ast/functions.go:524 +0xd6

[2021-07-27T03:30:45.316Z] github.com/pingcap/tidb/planner/core.(*PlanBuilder).rewriteExprNode(0xc000628340, 0xc001584340, 0x4085178, 0xc00024a2a0, 0xc001584001, 0x0, 0x0, 0x0, 0x0, 0x0, ...)

[2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:198 +0x9c

[2021-07-27T03:30:45.316Z] github.com/pingcap/tidb/planner/core.(*PlanBuilder).rewriteWithPreprocess(0xc000628340, 0x40594e8, 0xc0000540b8, 0x4085178, 0xc00024a2a0, 0x409b010, 0xc001584000, 0x0, 0x0, 0x1, ...)

[2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:144 +0x14c

[2021-07-27T03:30:45.316Z] github.com/pingcap/tidb/planner/core.(*PlanBuilder).rewrite(0xc000628340, 0x40594e8, 0xc0000540b8, 0x4085178, 0xc00024a2a0, 0x409b010, 0xc001584000, 0x0, 0x1, 0x0, ...)

[2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:112 +0xb9

[2021-07-27T03:30:45.316Z] github.com/pingcap/tidb/planner/core.rewriteAstExpr(0x409d238, 0xc00057c080, 0x4085178, 0xc00024a2a0, 0xc0009900a0, 0xc00152ccc0, 0x6, 0x6, 0x3840c20, 0x7effdc67a201, ...)

[2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:79 +0x36c

[2021-07-27T03:30:45.316Z] github.com/pingcap/br/pkg/lightning/backend/kv.collectGeneratedColumns(0xc00057c080, 0xc000d8a300, 0xc000890c60, 0x6, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0)

[2021-07-27T03:30:45.316Z] 	/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/pkg/lightning/backend/kv/sql2kv.go:170 +0x7c2

[2021-07-27T03:30:45.316Z] github.com/pingcap/br/pkg/lightning/backend/kv.NewTableKVEncoder(0x408e8f0, 0xc0002a05a0, 0xc00110e040, 0x20, 0x387ca40, 0x1, 0xc00110e040)

[2021-07-27T03:30:45.316Z] 	/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/pkg/lightning/backend/kv/sql2kv.go:100 +0x4c5

[2021-07-27T03:30:45.316Z] github.com/pingcap/br/pkg/lightning/backend/local.(*local).NewEncoder(0xc0005421e0, 0x408e8f0, 0xc0002a05a0, 0xc00110e040, 0x0, 0x0, 0x0, 0x0)

[2021-07-27T03:30:45.316Z] 	/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/pkg/lightning/backend/local/local.go:2269 +0x45

[2021-07-27T03:30:45.316Z] github.com/pingcap/br/pkg/lightning/backend.Backend.NewEncoder(...)

[2021-07-27T03:30:45.317Z] 	/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/pkg/lightning/backend/backend.go:252

[2021-07-27T03:30:45.317Z] github.com/pingcap/br/pkg/lightning/restore.(*chunkRestore).restore(0xc00110ee20, 0x40594b0, 0xc00059eec0, 0xc000202460, 0x0, 0xc00110ee40, 0xc00110ee60, 0xc000f9e900, 0x0, 0x0)

[2021-07-27T03:30:45.317Z] 	/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/pkg/lightning/restore/restore.go:2276 +0xf7

[2021-07-27T03:30:45.317Z] github.com/pingcap/br/pkg/lightning/restore.(*TableRestore).restoreEngine.func2(0xc000380590, 0xc000f9e900, 0x3ff0000000000000, 0x40594b0, 0xc00059eec0, 0xc000202460, 0xc000000000, 0xc00110ee40, 0xc00110ee60, 0xc0003805a0, ...)

[2021-07-27T03:30:45.317Z] 	/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/pkg/lightning/restore/table_restore.go:495 +0x18c

[2021-07-27T03:30:45.317Z] created by github.com/pingcap/br/pkg/lightning/restore.(*TableRestore).restoreEngine

[2021-07-27T03:30:45.317Z] 	/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/pkg/lightning/restore/table_restore.go:487 +0x9c5
```

**Comment by AndreMouche** [comment]
> Find a panic in tests
> 
> [ci.pingcap.net/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/6172/pipeline#step-471-log-68](https://ci.pingcap.net/blue/organizations/jenkins/br_ghpr_unit_and_integration_test/detail/br_ghpr_unit_and_integration_test/6172/pipeline#step-471-log-68)
> 
> ```
> [2021-07-27T03:30:45.315Z] panic: runtime error: invalid memory address or nil pointer dereference
> 
> [2021-07-27T03:30:45.316Z] [signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x2eaded2]
> 
> [2021-07-27T03:30:45.316Z] 
> 
> [2021-07-27T03:30:45.316Z] goroutine 778 [running]:
> 
> [2021-07-27T03:30:45.316Z] github.com/pingcap/br/pkg/lightning/backend/kv.(*session).GetBuiltinFunctionUsage(0xc00057c080, 0xc00057c080)
> 
> [2021-07-27T03:30:45.316Z] 	<autogenerated>:1 +0x32
> 
> [2021-07-27T03:30:45.316Z] github.com/pingcap/tidb/planner/core.(*expressionRewriter).newFunction(0xc001584340, 0xc000fc4733, 0x4, 0xc00024a0f8, 0xc00064c040, 0x3, 0x4, 0xc000140fc8, 0x7effe4007558, 0x20, ...)
> 
> [2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:1194 +0x173
> 
> [2021-07-27T03:30:45.316Z] github.com/pingcap/tidb/planner/core.(*expressionRewriter).funcCallToExpression(0xc001584340, 0xc00024a0e0)
> 
> [2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:1752 +0x8db
> 
> [2021-07-27T03:30:45.316Z] github.com/pingcap/tidb/planner/core.(*expressionRewriter).Leave(0xc001584340, 0x406fdf0, 0xc00024a0e0, 0x40884a8, 0xc000c169c0, 0x1)
> 
> [2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:1064 +0x19ad
> 
> [2021-07-27T03:30:45.316Z] github.com/pingcap/parser/ast.(*FuncCallExpr).Accept(0xc00024a0e0, 0x40361b8, 0xc001584340, 0x406fdf0, 0xc00024a2a0, 0xc00168a000)
> 
> [2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/parser@v0.0.0-20210707071004-31c87e37af5c/ast/functions.go:530 +0x18e
> 
> [2021-07-27T03:30:45.316Z] github.com/pingcap/parser/ast.(*FuncCallExpr).Accept(0xc00024a2a0, 0x40361b8, 0xc001584340, 0x409b010, 0xc001584000, 0xc0001413e8)
> 
> [2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/parser@v0.0.0-20210707071004-31c87e37af5c/ast/functions.go:524 +0xd6
> 
> [2021-07-27T03:30:45.316Z] github.com/pingcap/tidb/planner/core.(*PlanBuilder).rewriteExprNode(0xc000628340, 0xc001584340, 0x4085178, 0xc00024a2a0, 0xc001584001, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
> 
> [2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:198 +0x9c
> 
> [2021-07-27T03:30:45.316Z] github.com/pingcap/tidb/planner/core.(*PlanBuilder).rewriteWithPreprocess(0xc000628340, 0x40594e8, 0xc0000540b8, 0x4085178, 0xc00024a2a0, 0x409b010, 0xc001584000, 0x0, 0x0, 0x1, ...)
> 
> [2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:144 +0x14c
> 
> [2021-07-27T03:30:45.316Z] github.com/pingcap/tidb/planner/core.(*PlanBuilder).rewrite(0xc000628340, 0x40594e8, 0xc0000540b8, 0x4085178, 0xc00024a2a0, 0x409b010, 0xc001584000, 0x0, 0x1, 0x0, ...)
> 
> [2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:112 +0xb9
> 
> [2021-07-27T03:30:45.316Z] github.com/pingcap/tidb/planner/core.rewriteAstExpr(0x409d238, 0xc00057c080, 0x4085178, 0xc00024a2a0, 0xc0009900a0, 0xc00152ccc0, 0x6, 0x6, 0x3840c20, 0x7effdc67a201, ...)
> 
> [2021-07-27T03:30:45.316Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:79 +0x36c
> 
> [2021-07-27T03:30:45.316Z] github.com/pingcap/br/pkg/lightning/backend/kv.collectGeneratedColumns(0xc00057c080, 0xc000d8a300, 0xc000890c60, 0x6, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0)
> 
> [2021-07-27T03:30:45.316Z] 	/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/pkg/lightning/backend/kv/sql2kv.go:170 +0x7c2
> 
> [2021-07-27T03:30:45.316Z] github.com/pingcap/br/pkg/lightning/backend/kv.NewTableKVEncoder(0x408e8f0, 0xc0002a05a0, 0xc00110e040, 0x20, 0x387ca40, 0x1, 0xc00110e040)
> 
> [2021-07-27T03:30:45.316Z] 	/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/pkg/lightning/backend/kv/sql2kv.go:100 +0x4c5
> 
> [2021-07-27T03:30:45.316Z] github.com/pingcap/br/pkg/lightning/backend/local.(*local).NewEncoder(0xc0005421e0, 0x408e8f0, 0xc0002a05a0, 0xc00110e040, 0x0, 0x0, 0x0, 0x0)
> 
> [2021-07-27T03:30:45.316Z] 	/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/pkg/lightning/backend/local/local.go:2269 +0x45
> 
> [2021-07-27T03:30:45.316Z] github.com/pingcap/br/pkg/lightning/backend.Backend.NewEncoder(...)
> 
> [2021-07-27T03:30:45.317Z] 	/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/pkg/lightning/backend/backend.go:252
> 
> [2021-07-27T03:30:45.317Z] github.com/pingcap/br/pkg/lightning/restore.(*chunkRestore).restore(0xc00110ee20, 0x40594b0, 0xc00059eec0, 0xc000202460, 0x0, 0xc00110ee40, 0xc00110ee60, 0xc000f9e900, 0x0, 0x0)
> 
> [2021-07-27T03:30:45.317Z] 	/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/pkg/lightning/restore/restore.go:2276 +0xf7
> 
> [2021-07-27T03:30:45.317Z] github.com/pingcap/br/pkg/lightning/restore.(*TableRestore).restoreEngine.func2(0xc000380590, 0xc000f9e900, 0x3ff0000000000000, 0x40594b0, 0xc00059eec0, 0xc000202460, 0xc000000000, 0xc00110ee40, 0xc00110ee60, 0xc0003805a0, ...)
> 
> [2021-07-27T03:30:45.317Z] 	/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/pkg/lightning/restore/table_restore.go:495 +0x18c
> 
> [2021-07-27T03:30:45.317Z] created by github.com/pingcap/br/pkg/lightning/restore.(*TableRestore).restoreEngine
> 
> [2021-07-27T03:30:45.317Z] 	/home/jenkins/agent/workspace/br_ghpr_unit_and_integration_test/go/src/github.com/pingcap/br/pkg/lightning/restore/table_restore.go:487 +0x9c5
> ```

I have no idea about this, any suggestion?

File: pkg/lightning/backend/kv/session.go [hunk 1]
```diff
@@ -319,6 +319,12 @@ func (se *session) GetInfoSchema() sessionctx.InfoschemaMetaVersion {
 	return nil
 }
 
+// GetBuiltinFunctionUsage returns the BuiltinFunctionUsage of current Context, which is not thread safe.
+// Use primitive map type to prevent circular import. Should convert it to telemetry.BuiltinFunctionUsage before using.
+func (se *session) GetBuiltinFunctionUsage() map[string]uint32 {
+	return nil
```
Comment (kennytm) [inline pkg/lightning/backend/kv/session.go:325]:
```suggestion
	return make(map[string]uint32)
```

```
[2021-07-27T08:10:59.943Z] panic: assignment to entry in nil map

[2021-07-27T08:10:59.943Z] 

[2021-07-27T08:10:59.943Z] goroutine 756 [running]:

[2021-07-27T08:10:59.943Z] github.com/pingcap/tidb/telemetry.BuiltinFunctionsUsage.Inc(...)

[2021-07-27T08:10:59.943Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/telemetry/data_window.go:127

[2021-07-27T08:10:59.943Z] github.com/pingcap/tidb/planner/core.(*expressionRewriter).newFunction(0xc000cd88f0, 0xc000ce3c33, 0x4, 0xc000822398, 0xc0006ba000, 0x3, 0x4, 0xc000c8afc8, 0x7f3120357778, 0x20, ...)

[2021-07-27T08:10:59.943Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:1194 +0x288

[2021-07-27T08:10:59.943Z] github.com/pingcap/tidb/planner/core.(*expressionRewriter).funcCallToExpression(0xc000cd88f0, 0xc000822380)

[2021-07-27T08:10:59.943Z] 	/nfs/cache/mod/github.com/pingcap/tidb@v0.0.0-20210727021217-150fe0c37755/planner/core/expression_rewriter.go:1752 +0x8db
```

**Review by kennytm — APPROVED** [review]

**Pull request merged**